{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# performing RAxML analysis: \n",
    "will have to rename files and specify so that rax object recognizes files already made. \n",
    "default name for rax object is \"test\"\n",
    "\n",
    "NOTE: will likely have to give actual file location string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ipyrad.analysis as ipa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'raxmlHPC-PTHREADS-SSE3 -f a -T 4 -m GTRGAMMA -N 100 -x 12345 -p 54321 -n test -w /rigel/home/moh2112/slurm-logs/analysis-raxml -s /rigel/home/moh2112/scratch-dsi/DataAnalysisandNotebooks/quercus/quercus2_outfiles/quercus2.phy'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rax = ipa.raxml(data=\"../scratch-dsi/DataAnalysisandNotebooks/quercus/quercus2_outfiles/quercus2.phy\")\n",
    "rax.command"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rax.run(force=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "Qrobdip = open(\"Q.roburGenome/Qrob_diploid_version_2.fa\")\n",
    "Qrobhap = open(\"Q.roburGenome/Qrob_haploid.fa\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Downloading the Sequence Data\n",
    "\n",
    "Sequence data are from Eaton et al. 2015. The dataset includes *Quercus* ser *Virentes* sequence reads, as well as outgroup samples. The accession number used for downloading the sra files: 'SRP055977'."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "IPyradWarningExit",
     "evalue": "\nTo use the ipa.sratools module you must install two additional \nlibraries which can be done with the following conda command. \n\n  conda install -c bioconda sra-tools entrez-direct\n",
     "output_type": "error",
     "traceback": [
      "An exception has occurred, use %tb to see the full traceback.\n",
      "\u001b[0;31mIPyradWarningExit\u001b[0m\u001b[0;31m:\u001b[0m \nTo use the ipa.sratools module you must install two additional \nlibraries which can be done with the following conda command. \n\n  conda install -c bioconda sra-tools entrez-direct\n\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/rigel/home/moh2112/miniconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2969: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.\n",
      "  warn(\"To exit: use 'exit', 'quit', or Ctrl-D.\", stacklevel=1)\n"
     ]
    }
   ],
   "source": [
    "import ipyrad.analysis as ipa\n",
    "\n",
    "sra = ipa.sratools(accession=\"SRP055977\", workdir=\"/rigel/dsi/users/moh2112/DataAnalysisandNotebooks/rawdata/\")\n",
    "\n",
    "# NOTE: not sure why it's not working now... these libraries have been installed for a while"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'sra' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-46-fea48d42604e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# from: http://eaton-lab.org/posts/sra-downloads/\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;31m# 'fetching' the SRA project assemblies\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mproj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msra\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfetch_runinfo\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'sra' is not defined"
     ]
    }
   ],
   "source": [
    "# from: http://eaton-lab.org/posts/sra-downloads/\n",
    "# 'fetching' the SRA project assemblies\n",
    "proj = sra.fetch_runinfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[skip] file already exists: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/AR_re_SRR1915524.fastq.gz\n",
      "[skip] file already exists: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/BJSB3_v_SRR1915538.fastq.gz\n",
      "[skip] file already exists: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/BJSL25_re_SRR1915525.fastq.gz\n",
      "[skip] file already exists: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/BJSL25_v_SRR1915539.fastq.gz\n",
      "[skip] file already exists: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/BJVL19_re_SRR1915526.fastq.gz\n",
      "[skip] file already exists: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/BJVL19_v_SRR1915540.fastq.gz\n",
      "[skip] file already exists: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/BZBB1_v_SRR1915541.fastq.gz\n",
      "Downloading file 8/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CH_re_SRR1915527.fastq.gz\n",
      "Downloading file 9/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CRL0001_re_SRR1915528.fastq.gz\n",
      "Downloading file 10/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CRL0001_v_SRR1915542.fastq.gz\n",
      "Downloading file 11/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CRL0030_re_SRR1915529.fastq.gz\n",
      "Downloading file 12/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CRL0030_v_SRR1915543.fastq.gz\n",
      "Downloading file 13/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CUCA4_v_SRR1915544.fastq.gz\n",
      "Downloading file 14/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CUMM5_v_SRR1915545.fastq.gz\n",
      "Downloading file 15/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CUSV6_v_SRR1915546.fastq.gz\n",
      "Downloading file 16/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/CUVN10_v_SRR1915547.fastq.gz\n",
      "Downloading file 17/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/DO_re_SRR1915530.fastq.gz\n",
      "Downloading file 18/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/DU_re_SRR1915531.fastq.gz\n",
      "Downloading file 19/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/EN_re_SRR1915532.fastq.gz\n",
      "Downloading file 20/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLAB109_re_SRR1915533.fastq.gz\n",
      "Downloading file 21/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLAB109_v_SRR1915548.fastq.gz\n",
      "Downloading file 22/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLBA140_re_SRR1915534.fastq.gz\n",
      "Downloading file 23/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLBA140_v_SRR1915549.fastq.gz\n",
      "Downloading file 24/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLCK18_v_SRR1915550.fastq.gz\n",
      "Downloading file 25/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLCK216_v_SRR1915551.fastq.gz\n",
      "Downloading file 26/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLMO62_v_SRR1915552.fastq.gz\n",
      "Downloading file 27/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLSA185_v_SRR1915553.fastq.gz\n",
      "Downloading file 28/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLSF33_v_SRR1915554.fastq.gz\n",
      "Downloading file 29/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLSF47_v_SRR1915555.fastq.gz\n",
      "Downloading file 30/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLSF54_re_SRR1915535.fastq.gz\n",
      "Downloading file 31/43: /Users/maiahernandez/Desktop/Senior Thesis/Data Analysis and Notebooks/rawdata/FLSF54_v_SRR1915556.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "# download the raw data files. Files were downloaded into a file named 'rawdata'\n",
    "sra.run(name_fields=(30,1), name_separator=\"_\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Assemble the Data\n",
    "The data need to be put together and placed on the reference genome for *Q. robur*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New Assembly: Quercus\n"
     ]
    }
   ],
   "source": [
    "import ipyrad as ip\n",
    "data = ip.Assembly('Quercus') # create an Assembly object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0   assembly_name               Quercus                                      \n",
      "1   project_dir                 /rigel/home/moh2112/scratch-dsi/DataAnalysisandNotebooks/quercus/\n",
      "2   raw_fastq_path                                                           \n",
      "3   barcodes_path                                                            \n",
      "4   sorted_fastq_path           /rigel/home/moh2112/scratch-dsi/rawdata/*.gz \n",
      "5   assembly_method             denovo                                       \n",
      "6   reference_sequence                                                       \n",
      "7   datatype                    rad                                          \n",
      "8   restriction_overhang        ('TGCAG', '')                                \n",
      "9   max_low_qual_bases          5                                            \n",
      "10  phred_Qscore_offset         33                                           \n",
      "11  mindepth_statistical        5                                            \n",
      "12  mindepth_majrule            2                                            \n",
      "13  maxdepth                    10000                                        \n",
      "14  clust_threshold             0.85                                         \n",
      "15  max_barcode_mismatch        0                                            \n",
      "16  filter_adapters             0                                            \n",
      "17  filter_min_trim_len         35                                           \n",
      "18  max_alleles_consens         2                                            \n",
      "19  max_Ns_consens              (5, 5)                                       \n",
      "20  max_Hs_consens              (8, 8)                                       \n",
      "21  min_samples_locus           4                                            \n",
      "22  max_SNPs_locus              (20, 20)                                     \n",
      "23  max_Indels_locus            (8, 8)                                       \n",
      "24  max_shared_Hs_locus         0.5                                          \n",
      "25  trim_reads                  (0, 0, 0, 0)                                 \n",
      "26  trim_loci                   (0, 0, 0, 0)                                 \n",
      "27  output_formats              ('l', 'p', 's', 'v', 'a', 'k', 'u')          \n",
      "28  pop_assign_file                                                          \n"
     ]
    }
   ],
   "source": [
    "# 'data.set_params('project_dir', 'assembly')' giving errors, saying there is a space in the name...\n",
    "\n",
    "data.set_params('sorted_fastq_path', '~/scratch-dsi/rawdata/*.gz')\n",
    "data.set_params('mindepth_statistical', '5')\n",
    "data.set_params('mindepth_majrule', '2')\n",
    "data.set_params('project_dir', '~/scratch-dsi/DataAnalysisandNotebooks/quercus/')\n",
    "data.set_params('output_formats', 'lpsvaku')\n",
    "# data.set_params('output_formats', 'nlpsvaku') or *?\n",
    "data.get_params()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Assembly: Quercus\n",
      "[####################] 100% 0:00:11 | loading reads        | s1 |\n",
      "[####################] 100% 0:04:38 | processing reads     | s2 |\n"
     ]
    }
   ],
   "source": [
    "data.run('12', force=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Assembly: Quercus\n",
      "[####################] 100% 0:00:01 | concatenating        | s3 |\n",
      "[####################] 100% 0:00:24 | dereplicating        | s3 |\n",
      "[####################] 100% 0:10:01 | clustering/mapping   | s3 |\n",
      "[####################] 100% 0:00:01 | building clusters    | s3 |\n",
      "[####################] 100% 0:00:00 | chunking clusters    | s3 |\n",
      "[####################] 100% 0:19:51 | aligning clusters    | s3 |\n",
      "[####################] 100% 0:00:10 | concat clusters      | s3 |\n",
      "[####################] 100% 0:00:02 | calc cluster stats   | s3 |\n",
      "[####################] 100% 0:01:40 | inferring [H, E]     | s4 |\n",
      "[####################] 100% 0:00:02 | calculating depths   | s5 |\n",
      "[####################] 100% 0:00:04 | chunking clusters    | s5 |\n",
      "[####################] 100% 0:10:28 | consens calling      | s5 |\n",
      "[####################] 100% 0:00:25 | indexing alleles     | s5 |\n",
      "[####################] 100% 0:00:22 | concatenating inputs | s6 |\n",
      "[####################] 100% 0:32:40 | clustering tier 1    | s6 |\n",
      "[####################] 100% 0:00:03 | concatenating inputs | s6 |\n",
      "[####################] 100% 0:27:14 | clustering across    | s6 |\n",
      "[####################] 100% 0:00:21 | building clusters    | s6 |\n",
      "[####################] 100% 0:04:26 | aligning clusters    | s6 |\n"
     ]
    }
   ],
   "source": [
    "data.run(3456)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0   assembly_name               Quercus                                      \n",
      "1   project_dir                 /rigel/home/moh2112/scratch-dsi/DataAnalysisandNotebooks/quercus/\n",
      "2   raw_fastq_path                                                           \n",
      "3   barcodes_path                                                            \n",
      "4   sorted_fastq_path           /rigel/home/moh2112/scratch-dsi/rawdata/*.gz \n",
      "5   assembly_method             denovo                                       \n",
      "6   reference_sequence                                                       \n",
      "7   datatype                    rad                                          \n",
      "8   restriction_overhang        ('TGCAG', '')                                \n",
      "9   max_low_qual_bases          5                                            \n",
      "10  phred_Qscore_offset         33                                           \n",
      "11  mindepth_statistical        5                                            \n",
      "12  mindepth_majrule            2                                            \n",
      "13  maxdepth                    10000                                        \n",
      "14  clust_threshold             0.85                                         \n",
      "15  max_barcode_mismatch        0                                            \n",
      "16  filter_adapters             0                                            \n",
      "17  filter_min_trim_len         35                                           \n",
      "18  max_alleles_consens         2                                            \n",
      "19  max_Ns_consens              (5, 5)                                       \n",
      "20  max_Hs_consens              (8, 8)                                       \n",
      "21  min_samples_locus           4                                            \n",
      "22  max_SNPs_locus              (20, 20)                                     \n",
      "23  max_Indels_locus            (8, 8)                                       \n",
      "24  max_shared_Hs_locus         0.5                                          \n",
      "25  trim_reads                  (0, 0, 0, 0)                                 \n",
      "26  trim_loci                   (0, 0, 0, 0)                                 \n",
      "27  output_formats              ('l', 'p', 's', 'v', 'a', 'k', 'u')          \n",
      "28  pop_assign_file                                                          \n"
     ]
    }
   ],
   "source": [
    "data.set_params('output_formats', 'lpsvaku')\n",
    "\n",
    "data.get_params()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Assembly: Quercus\n",
      "[####################] 100% 0:00:17 | applying filters     | s7 |\n"
     ]
    }
   ],
   "source": [
    "data.run('7')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# after step 7\n",
    "# data.outfiles.phy --> raxml\n",
    "# data.outfiles.loci --> visual inspection of the data, and abba-baba\n",
    "# data.outfiles.struct --> structure analysis\n",
    "# data.outfiles.alleles --> bucky, bpp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# After assembling the data:\n",
    "load .json file to begin using the analysis tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ipyrad as ip\n",
    "\n",
    "import ipyrad.analysis as ipa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loading Assembly: Quercus\n",
      "from saved path: ~/scratch-dsi/DataAnalysisandNotebooks/quercus/Quercus.json\n"
     ]
    }
   ],
   "source": [
    "data1 = ip.load_json('~/scratch-dsi/DataAnalysisandNotebooks/quercus/Quercus.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0   assembly_name               quercus2                                     \n",
      "1   project_dir                 /rigel/home/moh2112/scratch-dsi/DataAnalysisandNotebooks/quercus\n",
      "2   raw_fastq_path                                                           \n",
      "3   barcodes_path                                                            \n",
      "4   sorted_fastq_path           /rigel/home/moh2112/scratch-dsi/rawdata/*.gz \n",
      "5   assembly_method             denovo                                       \n",
      "6   reference_sequence                                                       \n",
      "7   datatype                    rad                                          \n",
      "8   restriction_overhang        ('TGCAG', '')                                \n",
      "9   max_low_qual_bases          5                                            \n",
      "10  phred_Qscore_offset         33                                           \n",
      "11  mindepth_statistical        5                                            \n",
      "12  mindepth_majrule            2                                            \n",
      "13  maxdepth                    10000                                        \n",
      "14  clust_threshold             0.85                                         \n",
      "15  max_barcode_mismatch        0                                            \n",
      "16  filter_adapters             0                                            \n",
      "17  filter_min_trim_len         35                                           \n",
      "18  max_alleles_consens         2                                            \n",
      "19  max_Ns_consens              (5, 5)                                       \n",
      "20  max_Hs_consens              (8, 8)                                       \n",
      "21  min_samples_locus           4                                            \n",
      "22  max_SNPs_locus              (20, 20)                                     \n",
      "23  max_Indels_locus            (8, 8)                                       \n",
      "24  max_shared_Hs_locus         0.5                                          \n",
      "25  trim_reads                  (0, 0, 0, 0)                                 \n",
      "26  trim_loci                   (0, 0, 0, 0)                                 \n",
      "27  output_formats              ('l', 'p', 's', 'v', 'a', 'k', 'u')          \n",
      "28  pop_assign_file                                                          \n"
     ]
    }
   ],
   "source": [
    "# September 20, 2018: ipyrad updated on September 18th and assembly steps 5-7 had to be re-run\n",
    "# because the assembly was already run on a previous version of ipyrad, the \n",
    "data2 = data1.branch(\"quercus2\")\n",
    "\n",
    "data2.set_params(\"project_dir\", \"/rigel/home/moh2112/scratch-dsi/DataAnalysisandNotebooks/quercus\")\n",
    "data2.get_params()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'data2' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-111355d2d07b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdata2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'567'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mforce\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'data2' is not defined"
     ]
    }
   ],
   "source": [
    "data2.run('567',force=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loading Assembly: quercus2\n",
      "from saved path: ~/scratch-dsi/DataAnalysisandNotebooks/quercus/quercus2.json\n"
     ]
    }
   ],
   "source": [
    "# reload the assembled .json file. quercus2.json is the most recent assembly as of September 18th\n",
    "data2 = ip.load_json(\"/rigel/home/moh2112/scratch-dsi/DataAnalysisandNotebooks/quercus/quercus2.json\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/rigel/home/moh2112/scratch-dsi/DataAnalysisandNotebooks/quercus/quercus2_outfiles/quercus2.phy'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# data2.outfiles.phy\n",
    "data2.outfiles.phy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>state</th>\n",
       "      <th>reads_raw</th>\n",
       "      <th>reads_passed_filter</th>\n",
       "      <th>clusters_total</th>\n",
       "      <th>clusters_hidepth</th>\n",
       "      <th>hetero_est</th>\n",
       "      <th>error_est</th>\n",
       "      <th>reads_consens</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AR_re_SRR1915524</th>\n",
       "      <td>6</td>\n",
       "      <td>4046890</td>\n",
       "      <td>4045035</td>\n",
       "      <td>105419</td>\n",
       "      <td>87652</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.002</td>\n",
       "      <td>82609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BJSB3_v_SRR1915538</th>\n",
       "      <td>6</td>\n",
       "      <td>931926</td>\n",
       "      <td>930900</td>\n",
       "      <td>82337</td>\n",
       "      <td>70567</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.002</td>\n",
       "      <td>67162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BJSL25_re_SRR1915525</th>\n",
       "      <td>6</td>\n",
       "      <td>5352627</td>\n",
       "      <td>5350196</td>\n",
       "      <td>114428</td>\n",
       "      <td>93763</td>\n",
       "      <td>0.017</td>\n",
       "      <td>0.002</td>\n",
       "      <td>88142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BJSL25_v_SRR1915539</th>\n",
       "      <td>6</td>\n",
       "      <td>969575</td>\n",
       "      <td>968555</td>\n",
       "      <td>84471</td>\n",
       "      <td>72713</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.002</td>\n",
       "      <td>69217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BJVL19_re_SRR1915526</th>\n",
       "      <td>6</td>\n",
       "      <td>4715624</td>\n",
       "      <td>4713397</td>\n",
       "      <td>108754</td>\n",
       "      <td>89643</td>\n",
       "      <td>0.017</td>\n",
       "      <td>0.002</td>\n",
       "      <td>84598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BJVL19_v_SRR1915540</th>\n",
       "      <td>6</td>\n",
       "      <td>817443</td>\n",
       "      <td>816492</td>\n",
       "      <td>78789</td>\n",
       "      <td>67901</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.002</td>\n",
       "      <td>64728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BZBB1_v_SRR1915541</th>\n",
       "      <td>6</td>\n",
       "      <td>849191</td>\n",
       "      <td>848308</td>\n",
       "      <td>96370</td>\n",
       "      <td>79450</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.003</td>\n",
       "      <td>75819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CH_re_SRR1915527</th>\n",
       "      <td>6</td>\n",
       "      <td>4539385</td>\n",
       "      <td>4537274</td>\n",
       "      <td>110414</td>\n",
       "      <td>90888</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.002</td>\n",
       "      <td>85430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRL0001_re_SRR1915528</th>\n",
       "      <td>6</td>\n",
       "      <td>3742953</td>\n",
       "      <td>3741455</td>\n",
       "      <td>153172</td>\n",
       "      <td>123844</td>\n",
       "      <td>0.019</td>\n",
       "      <td>0.002</td>\n",
       "      <td>116991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRL0001_v_SRR1915542</th>\n",
       "      <td>6</td>\n",
       "      <td>1012884</td>\n",
       "      <td>1011750</td>\n",
       "      <td>111076</td>\n",
       "      <td>88197</td>\n",
       "      <td>0.021</td>\n",
       "      <td>0.002</td>\n",
       "      <td>84555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRL0030_re_SRR1915529</th>\n",
       "      <td>6</td>\n",
       "      <td>3242503</td>\n",
       "      <td>3241207</td>\n",
       "      <td>165732</td>\n",
       "      <td>131339</td>\n",
       "      <td>0.015</td>\n",
       "      <td>0.002</td>\n",
       "      <td>124089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRL0030_v_SRR1915543</th>\n",
       "      <td>6</td>\n",
       "      <td>1204952</td>\n",
       "      <td>1203606</td>\n",
       "      <td>115844</td>\n",
       "      <td>91333</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.002</td>\n",
       "      <td>87446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUCA4_v_SRR1915544</th>\n",
       "      <td>6</td>\n",
       "      <td>570148</td>\n",
       "      <td>569523</td>\n",
       "      <td>79685</td>\n",
       "      <td>65595</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.003</td>\n",
       "      <td>62598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUMM5_v_SRR1915545</th>\n",
       "      <td>6</td>\n",
       "      <td>90661</td>\n",
       "      <td>90565</td>\n",
       "      <td>26748</td>\n",
       "      <td>19263</td>\n",
       "      <td>0.013</td>\n",
       "      <td>0.002</td>\n",
       "      <td>18734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUSV6_v_SRR1915546</th>\n",
       "      <td>6</td>\n",
       "      <td>775357</td>\n",
       "      <td>774521</td>\n",
       "      <td>83272</td>\n",
       "      <td>70813</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.003</td>\n",
       "      <td>67345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUVN10_v_SRR1915547</th>\n",
       "      <td>6</td>\n",
       "      <td>1130911</td>\n",
       "      <td>1129633</td>\n",
       "      <td>92568</td>\n",
       "      <td>78579</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.002</td>\n",
       "      <td>74644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DO_re_SRR1915530</th>\n",
       "      <td>6</td>\n",
       "      <td>1953553</td>\n",
       "      <td>1952913</td>\n",
       "      <td>90620</td>\n",
       "      <td>78518</td>\n",
       "      <td>0.019</td>\n",
       "      <td>0.002</td>\n",
       "      <td>74307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DU_re_SRR1915531</th>\n",
       "      <td>6</td>\n",
       "      <td>3744449</td>\n",
       "      <td>3742756</td>\n",
       "      <td>110187</td>\n",
       "      <td>91770</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.002</td>\n",
       "      <td>86431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EN_re_SRR1915532</th>\n",
       "      <td>6</td>\n",
       "      <td>743556</td>\n",
       "      <td>743319</td>\n",
       "      <td>72221</td>\n",
       "      <td>61842</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.003</td>\n",
       "      <td>58963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLAB109_re_SRR1915533</th>\n",
       "      <td>6</td>\n",
       "      <td>3219505</td>\n",
       "      <td>3218039</td>\n",
       "      <td>232463</td>\n",
       "      <td>192673</td>\n",
       "      <td>0.012</td>\n",
       "      <td>0.002</td>\n",
       "      <td>185711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLAB109_v_SRR1915548</th>\n",
       "      <td>6</td>\n",
       "      <td>347949</td>\n",
       "      <td>347605</td>\n",
       "      <td>85056</td>\n",
       "      <td>62655</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.003</td>\n",
       "      <td>60713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLBA140_re_SRR1915534</th>\n",
       "      <td>6</td>\n",
       "      <td>2826213</td>\n",
       "      <td>2825251</td>\n",
       "      <td>111094</td>\n",
       "      <td>93917</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.002</td>\n",
       "      <td>89050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLBA140_v_SRR1915549</th>\n",
       "      <td>6</td>\n",
       "      <td>975393</td>\n",
       "      <td>974280</td>\n",
       "      <td>88007</td>\n",
       "      <td>74551</td>\n",
       "      <td>0.021</td>\n",
       "      <td>0.002</td>\n",
       "      <td>70952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLCK18_v_SRR1915550</th>\n",
       "      <td>6</td>\n",
       "      <td>941880</td>\n",
       "      <td>940882</td>\n",
       "      <td>123135</td>\n",
       "      <td>99118</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.003</td>\n",
       "      <td>94574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLCK216_v_SRR1915551</th>\n",
       "      <td>6</td>\n",
       "      <td>791393</td>\n",
       "      <td>790499</td>\n",
       "      <td>97521</td>\n",
       "      <td>79888</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.003</td>\n",
       "      <td>76232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLMO62_v_SRR1915552</th>\n",
       "      <td>6</td>\n",
       "      <td>905151</td>\n",
       "      <td>904157</td>\n",
       "      <td>86669</td>\n",
       "      <td>73310</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.003</td>\n",
       "      <td>69363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLSA185_v_SRR1915553</th>\n",
       "      <td>6</td>\n",
       "      <td>3976506</td>\n",
       "      <td>3972221</td>\n",
       "      <td>176251</td>\n",
       "      <td>134571</td>\n",
       "      <td>0.015</td>\n",
       "      <td>0.003</td>\n",
       "      <td>128766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLSF33_v_SRR1915554</th>\n",
       "      <td>6</td>\n",
       "      <td>857771</td>\n",
       "      <td>856850</td>\n",
       "      <td>92840</td>\n",
       "      <td>77058</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.003</td>\n",
       "      <td>73319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLSF47_v_SRR1915555</th>\n",
       "      <td>6</td>\n",
       "      <td>908978</td>\n",
       "      <td>908015</td>\n",
       "      <td>90066</td>\n",
       "      <td>75585</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.003</td>\n",
       "      <td>71785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLSF54_re_SRR1915535</th>\n",
       "      <td>6</td>\n",
       "      <td>2543324</td>\n",
       "      <td>2542473</td>\n",
       "      <td>101148</td>\n",
       "      <td>86399</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.002</td>\n",
       "      <td>81439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLSF54_v_SRR1915556</th>\n",
       "      <td>6</td>\n",
       "      <td>1115926</td>\n",
       "      <td>1114691</td>\n",
       "      <td>92903</td>\n",
       "      <td>79170</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.002</td>\n",
       "      <td>75283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLWO6_v_SRR1915557</th>\n",
       "      <td>6</td>\n",
       "      <td>1099920</td>\n",
       "      <td>1098781</td>\n",
       "      <td>124827</td>\n",
       "      <td>103880</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.003</td>\n",
       "      <td>97847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HE_re_SRR1915536</th>\n",
       "      <td>6</td>\n",
       "      <td>2777404</td>\n",
       "      <td>2776375</td>\n",
       "      <td>95049</td>\n",
       "      <td>81246</td>\n",
       "      <td>0.016</td>\n",
       "      <td>0.002</td>\n",
       "      <td>77196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HNDA09_v_SRR1915558</th>\n",
       "      <td>6</td>\n",
       "      <td>1112292</td>\n",
       "      <td>1111089</td>\n",
       "      <td>89381</td>\n",
       "      <td>76603</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.002</td>\n",
       "      <td>72539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LALC2_v_SRR1915559</th>\n",
       "      <td>6</td>\n",
       "      <td>1436228</td>\n",
       "      <td>1434611</td>\n",
       "      <td>102672</td>\n",
       "      <td>85839</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.002</td>\n",
       "      <td>81595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MXED8_v_SRR1915560</th>\n",
       "      <td>6</td>\n",
       "      <td>1154841</td>\n",
       "      <td>1153571</td>\n",
       "      <td>111029</td>\n",
       "      <td>89911</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.002</td>\n",
       "      <td>85551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MXGT4_v_SRR1915561</th>\n",
       "      <td>6</td>\n",
       "      <td>1347187</td>\n",
       "      <td>1345681</td>\n",
       "      <td>107231</td>\n",
       "      <td>88687</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.002</td>\n",
       "      <td>84309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MXSA3017_v_SRR1915562</th>\n",
       "      <td>6</td>\n",
       "      <td>1046574</td>\n",
       "      <td>1045454</td>\n",
       "      <td>118844</td>\n",
       "      <td>94939</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.003</td>\n",
       "      <td>90738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NI_re_SRR1915537</th>\n",
       "      <td>6</td>\n",
       "      <td>4482353</td>\n",
       "      <td>4480315</td>\n",
       "      <td>106624</td>\n",
       "      <td>87595</td>\n",
       "      <td>0.017</td>\n",
       "      <td>0.002</td>\n",
       "      <td>82586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SCCU3_v_SRR1915563</th>\n",
       "      <td>6</td>\n",
       "      <td>505504</td>\n",
       "      <td>504971</td>\n",
       "      <td>99084</td>\n",
       "      <td>75647</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.003</td>\n",
       "      <td>72961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TXGR3_v_SRR1915564</th>\n",
       "      <td>6</td>\n",
       "      <td>879229</td>\n",
       "      <td>878311</td>\n",
       "      <td>89498</td>\n",
       "      <td>74122</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.002</td>\n",
       "      <td>70717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TXMD3_v_SRR1915565</th>\n",
       "      <td>6</td>\n",
       "      <td>934386</td>\n",
       "      <td>933366</td>\n",
       "      <td>90229</td>\n",
       "      <td>75503</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.002</td>\n",
       "      <td>71855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TXWV2_v_SRR1915566</th>\n",
       "      <td>6</td>\n",
       "      <td>163427</td>\n",
       "      <td>163232</td>\n",
       "      <td>40285</td>\n",
       "      <td>29944</td>\n",
       "      <td>0.013</td>\n",
       "      <td>0.002</td>\n",
       "      <td>29089</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       state  reads_raw  reads_passed_filter  clusters_total  \\\n",
       "AR_re_SRR1915524           6    4046890              4045035          105419   \n",
       "BJSB3_v_SRR1915538         6     931926               930900           82337   \n",
       "BJSL25_re_SRR1915525       6    5352627              5350196          114428   \n",
       "BJSL25_v_SRR1915539        6     969575               968555           84471   \n",
       "BJVL19_re_SRR1915526       6    4715624              4713397          108754   \n",
       "BJVL19_v_SRR1915540        6     817443               816492           78789   \n",
       "BZBB1_v_SRR1915541         6     849191               848308           96370   \n",
       "CH_re_SRR1915527           6    4539385              4537274          110414   \n",
       "CRL0001_re_SRR1915528      6    3742953              3741455          153172   \n",
       "CRL0001_v_SRR1915542       6    1012884              1011750          111076   \n",
       "CRL0030_re_SRR1915529      6    3242503              3241207          165732   \n",
       "CRL0030_v_SRR1915543       6    1204952              1203606          115844   \n",
       "CUCA4_v_SRR1915544         6     570148               569523           79685   \n",
       "CUMM5_v_SRR1915545         6      90661                90565           26748   \n",
       "CUSV6_v_SRR1915546         6     775357               774521           83272   \n",
       "CUVN10_v_SRR1915547        6    1130911              1129633           92568   \n",
       "DO_re_SRR1915530           6    1953553              1952913           90620   \n",
       "DU_re_SRR1915531           6    3744449              3742756          110187   \n",
       "EN_re_SRR1915532           6     743556               743319           72221   \n",
       "FLAB109_re_SRR1915533      6    3219505              3218039          232463   \n",
       "FLAB109_v_SRR1915548       6     347949               347605           85056   \n",
       "FLBA140_re_SRR1915534      6    2826213              2825251          111094   \n",
       "FLBA140_v_SRR1915549       6     975393               974280           88007   \n",
       "FLCK18_v_SRR1915550        6     941880               940882          123135   \n",
       "FLCK216_v_SRR1915551       6     791393               790499           97521   \n",
       "FLMO62_v_SRR1915552        6     905151               904157           86669   \n",
       "FLSA185_v_SRR1915553       6    3976506              3972221          176251   \n",
       "FLSF33_v_SRR1915554        6     857771               856850           92840   \n",
       "FLSF47_v_SRR1915555        6     908978               908015           90066   \n",
       "FLSF54_re_SRR1915535       6    2543324              2542473          101148   \n",
       "FLSF54_v_SRR1915556        6    1115926              1114691           92903   \n",
       "FLWO6_v_SRR1915557         6    1099920              1098781          124827   \n",
       "HE_re_SRR1915536           6    2777404              2776375           95049   \n",
       "HNDA09_v_SRR1915558        6    1112292              1111089           89381   \n",
       "LALC2_v_SRR1915559         6    1436228              1434611          102672   \n",
       "MXED8_v_SRR1915560         6    1154841              1153571          111029   \n",
       "MXGT4_v_SRR1915561         6    1347187              1345681          107231   \n",
       "MXSA3017_v_SRR1915562      6    1046574              1045454          118844   \n",
       "NI_re_SRR1915537           6    4482353              4480315          106624   \n",
       "SCCU3_v_SRR1915563         6     505504               504971           99084   \n",
       "TXGR3_v_SRR1915564         6     879229               878311           89498   \n",
       "TXMD3_v_SRR1915565         6     934386               933366           90229   \n",
       "TXWV2_v_SRR1915566         6     163427               163232           40285   \n",
       "\n",
       "                       clusters_hidepth  hetero_est  error_est  reads_consens  \n",
       "AR_re_SRR1915524                  87652       0.018      0.002          82609  \n",
       "BJSB3_v_SRR1915538                70567       0.022      0.002          67162  \n",
       "BJSL25_re_SRR1915525              93763       0.017      0.002          88142  \n",
       "BJSL25_v_SRR1915539               72713       0.022      0.002          69217  \n",
       "BJVL19_re_SRR1915526              89643       0.017      0.002          84598  \n",
       "BJVL19_v_SRR1915540               67901       0.022      0.002          64728  \n",
       "BZBB1_v_SRR1915541                79450       0.023      0.003          75819  \n",
       "CH_re_SRR1915527                  90888       0.018      0.002          85430  \n",
       "CRL0001_re_SRR1915528            123844       0.019      0.002         116991  \n",
       "CRL0001_v_SRR1915542              88197       0.021      0.002          84555  \n",
       "CRL0030_re_SRR1915529            131339       0.015      0.002         124089  \n",
       "CRL0030_v_SRR1915543              91333       0.022      0.002          87446  \n",
       "CUCA4_v_SRR1915544                65595       0.025      0.003          62598  \n",
       "CUMM5_v_SRR1915545                19263       0.013      0.002          18734  \n",
       "CUSV6_v_SRR1915546                70813       0.024      0.003          67345  \n",
       "CUVN10_v_SRR1915547               78579       0.023      0.002          74644  \n",
       "DO_re_SRR1915530                  78518       0.019      0.002          74307  \n",
       "DU_re_SRR1915531                  91770       0.018      0.002          86431  \n",
       "EN_re_SRR1915532                  61842       0.020      0.003          58963  \n",
       "FLAB109_re_SRR1915533            192673       0.012      0.002         185711  \n",
       "FLAB109_v_SRR1915548              62655       0.023      0.003          60713  \n",
       "FLBA140_re_SRR1915534             93917       0.018      0.002          89050  \n",
       "FLBA140_v_SRR1915549              74551       0.021      0.002          70952  \n",
       "FLCK18_v_SRR1915550               99118       0.024      0.003          94574  \n",
       "FLCK216_v_SRR1915551              79888       0.024      0.003          76232  \n",
       "FLMO62_v_SRR1915552               73310       0.025      0.003          69363  \n",
       "FLSA185_v_SRR1915553             134571       0.015      0.003         128766  \n",
       "FLSF33_v_SRR1915554               77058       0.024      0.003          73319  \n",
       "FLSF47_v_SRR1915555               75585       0.024      0.003          71785  \n",
       "FLSF54_re_SRR1915535              86399       0.018      0.002          81439  \n",
       "FLSF54_v_SRR1915556               79170       0.023      0.002          75283  \n",
       "FLWO6_v_SRR1915557               103880       0.024      0.003          97847  \n",
       "HE_re_SRR1915536                  81246       0.016      0.002          77196  \n",
       "HNDA09_v_SRR1915558               76603       0.023      0.002          72539  \n",
       "LALC2_v_SRR1915559                85839       0.022      0.002          81595  \n",
       "MXED8_v_SRR1915560                89911       0.024      0.002          85551  \n",
       "MXGT4_v_SRR1915561                88687       0.023      0.002          84309  \n",
       "MXSA3017_v_SRR1915562             94939       0.022      0.003          90738  \n",
       "NI_re_SRR1915537                  87595       0.017      0.002          82586  \n",
       "SCCU3_v_SRR1915563                75647       0.024      0.003          72961  \n",
       "TXGR3_v_SRR1915564                74122       0.023      0.002          70717  \n",
       "TXMD3_v_SRR1915565                75503       0.024      0.002          71855  \n",
       "TXWV2_v_SRR1915566                29944       0.013      0.002          29089  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data2.stats"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
